File: ThirtyFour.UI/ThirtyFour.UI.Tests/Table/TableTests.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using System;

namespace ThirtyFour.UI.Tests.Tests.Table
{
    using Table = ThirtyFour.UI.Table.Table;

    [TestClass]
    public class TableTests : BaseTestSuite
    {
        [TestMethod]
        public void TestHeaderCells()
        {
            Table table = new Table(driver.FindElement(By.ClassName("cart")));
            Assert.AreEqual(4, table.HeaderCells.Count);
        }

        [TestMethod]
        public void TestBodyRows()
        {
            Table table = new Table(driver.FindElement(By.ClassName("cart")));
            Assert.AreEqual(3, table.BodyRows.Count);
        }

        [TestMethod]
        public void TestGetCellValue()
        {
            Table table = new Table(driver.FindElement(By.ClassName("cart")));
            String cellValue = table.GetCellValue("Product", "First Test Product", "Subtotal");
            Assert.AreEqual("$9.99", cellValue);
        }
    }
}

File: java/src/test/java/com/github/danhumphrey/thirtyfour/ui/test/table/TableTests.java
package com.github.danhumphrey.thirtyfour.ui.test.table;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.IsEqual.equalTo;

import java.util.List;

import org.junit.jupiter.api.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

import com.github.danhumphrey.thirtyfour.ui.table.Table;
import com.github.danhumphrey.thirtyfour.ui.test.BaseTestSuite;

public class TableTests extends BaseTestSuite {

	
	@Test
	public void testGetHeaderCells() {
		WebDriver driver = getDriver();
		Table table = new Table(driver.findElement(By.className("cart")));
		List<WebElement> headercells = table.getHeaderCells();
		assertThat(headercells.size(), equalTo(4));
	}
	
	@Test
	public void testGetBodyRows() {
		WebDriver driver = getDriver();
		Table table = new Table(driver.findElement(By.className("cart")));
		List<WebElement> rows = table.getBodyRows();
		assertThat(rows.size(), equalTo(3));
	}
	
	@Test 
	public void testGetCellValue() {
		WebDriver driver = getDriver();
		Table table = new Table(driver.findElement(By.className("cart")));
		String value = table.getCellValue("Product", "First Test Product", "Subtotal");
		assertThat(value, equalTo("$9.99"));
	}
}
